backend-tests/backend-tests.hs

```diff
@@ -71,107 +71,7 @@ import Prometheus hiding (Timestamp)
 -- type IIInterface m = [Candid.candidFile|../src/internet_identity/internet_identity.did|]
 -- just yet, because haskell-candid doesn’t like init arguments to service, so
 -- for now we have to copy it
-type IIInterface m = [Candid.candid|
-type UserNumber = nat64;
-type PublicKey = blob;
-type CredentialId = blob;
-type DeviceAlias = text;
-type DeviceKey = PublicKey;
-type UserKey = PublicKey;
-type SessionKey = PublicKey;
-type FrontendHostname = text;
-type Timestamp = nat64;
-type Purpose = variant {
-    recovery;
-    authentication;
-};
-type KeyType = variant {
-    unknown;
-    platform;
-    cross_platform;
-    seed_phrase;
-};
-
-type DeviceData = record {
-  pubkey : DeviceKey;
-  alias : text;
-  credential_id : opt CredentialId;
-  purpose : Purpose;
-  key_type : KeyType;
-};
-
-type RegisterResponse = variant {
-  registered: record { user_number: UserNumber; };
-  canister_full;
-};
-
-type Delegation = record {
-  pubkey: SessionKey;
-  expiration: Timestamp;
-  targets: opt vec principal;
-};
-type SignedDelegation = record {
-  delegation: Delegation;
-  signature: blob;
-};
-type GetDelegationResponse = variant {
-  signed_delegation: SignedDelegation;
-  no_such_delegation;
-};
-
-type InternetIdentityStats = record {
-  users_registered: nat64;
-  assigned_user_number_range: record { nat64; nat64; };
-};
-
-type ProofOfWork = record {
-  timestamp : Timestamp;
-  nonce : nat64;
-};
-
-type HeaderField = record { text; text; };
-
-type HttpRequest = record {
-  method: text;
-  url: text;
-  headers: vec HeaderField;
-  body: blob;
-};
-
-type HttpResponse = record {
-  status_code: nat16;
-  headers: vec HeaderField;
-  body: blob;
-  streaming_strategy: opt StreamingStrategy;
-};
-
-type StreamingCallbackHttpResponse = record {
-  body: blob;
-  token: Token;
-};
-
-type Token = record {};
-
-type StreamingStrategy = variant {
-  Callback: record {
-    callback: func (Token) -> (StreamingCallbackHttpResponse) query;
-    token: Token;
-  };
-};
-service : {
-  register : (DeviceData, ProofOfWork) -> (RegisterResponse);
-  add : (UserNumber, DeviceData) -> ();
-  remove : (UserNumber, DeviceKey) -> ();
-  lookup : (UserNumber) -> (vec DeviceData) query;
-  get_principal : (UserNumber, FrontendHostname) -> (principal) query;
-  stats : () -> (InternetIdentityStats) query;
-
-  prepare_delegation : (UserNumber, FrontendHostname, SessionKey, opt nat64) -> (UserKey, Timestamp);
-  get_delegation: (UserNumber, FrontendHostname, SessionKey, Timestamp) -> (GetDelegationResponse) query;
-
-  http_request: (request: HttpRequest) -> (HttpResponse) query;
-}
-  |]
+type IIInterface m = [Candid.candidFile|../src/internet_identity/internet_identity.did|]
 
 -- Names for some of these types. Unfortunately requires copying
```

Comment:
I wonder why?  This is not the topic of this PR but if you happen to know please share.

Reply:
Haskell-candid allows importing .did files wholesome to get thr the service type, but not that type definitions therein. Maybe adding that would be a nice new feature for a leisurely hacking afternoon

Reply:
working on it : https://github.com/nomeata/haskell-candid/pull/18